In [1]:
!pip install faster_whisper

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.0/2.0 MB 31.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 33.5/33.5 MB 20.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 192.3/192.3 MB 7.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.2/13.2 MB 66.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.0/46.0 kB 1.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 86.8/86.8 kB 3.7 MB/s eta 0:00:00


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
import os
import time
import warnings
warnings.filterwarnings("ignore", category=UserWarning, module="huggingface_hub")
from faster_whisper import WhisperModel

# Basic model

In [13]:
audio_folder_path = "/content/drive/MyDrive/MAESTRIA/4 SEMESTRE/Deep Learning/data/audio/"

# 'tiny', 'base', 'small', 'medium', 'large'
model = WhisperModel("base", compute_type="float32")

audio_files = [f for f in os.listdir(audio_folder_path) if f.endswith('.mp3')]

start_time = time.time()

for audio_file in audio_files:
    audio_file_path = os.path.join(audio_folder_path, audio_file)
    segments, meta = model.transcribe(audio_file_path)
    # hotwords=['inteligencia artificial', 'deep learning', 'aprendizaje profundo', 'gradiente', 'derivadas']
    num_segments = len(list(segments))
    print(f"\nSegmentos generados para {audio_file}: {num_segments}")

    transcript_output_path = os.path.join(audio_folder_path, f"{os.path.splitext(audio_file)[0]}.txt")

    with open(transcript_output_path, "w") as file:
        for segment in segments:
            file.write(segment.text + "\n")
    print(meta)
    #print(f"Transcripción guardada exitosamente: {transcript_output_path}\n")

end_time = time.time()
execution_time = end_time - start_time
print("-"*30)
print(f"\nProceso de transcripción completado en {execution_time:.2f} segundos.")


Segmentos generados para Universidad Central - Deep Learning - Clase 2: Regularización de Redes Neuronales (2024-08-17 07:15 GMT-5).mp3: 1440
TranscriptionInfo(language='es', language_probability=0.9763216972351074, duration=4138.9106875, duration_after_vad=4138.9106875, all_language_probs=[('es', 0.9763216972351074), ('en', 0.009112867526710033), ('pt', 0.0026215941179543734), ('it', 0.0016427653608843684), ('gl', 0.0015417025424540043), ('fr', 0.0014084535650908947), ('ko', 0.0011659155134111643), ('ja', 0.000985389924608171), ('de', 0.0006312716868706048), ('eu', 0.0005810946458950639), ('ca', 0.0003552295675035566), ('pl', 0.00032324809581041336), ('id', 0.00031603631214238703), ('ru', 0.00030844315188005567), ('tr', 0.000274369289400056), ('la', 0.00025478089810349047), ('nn', 0.0002390634617768228), ('nl', 0.0002221244794782251), ('jw', 0.00017257120634894818), ('cy', 0.00016452638374175876), ('zh', 0.00015690369764342904), ('ta', 9.946496720658615e-05), ('haw', 9.1252775746397

# Basic model with arbitrary hyperparameter adjustment

In [19]:
audio_folder_path = "/content/drive/MyDrive/MAESTRIA/4 SEMESTRE/Deep Learning/data/audio/"

# 'tiny', 'base', 'small', 'medium', 'large'
model = WhisperModel("base", compute_type="float16")
# hotwords=['inteligencia artificial', 'deep learning', 'aprendizaje profundo', 'gradiente', 'derivadas']

audio_files = [f for f in os.listdir(audio_folder_path) if f.endswith('.mp3')]

start_time = time.time()

for audio_file in audio_files:
    audio_file_path = os.path.join(audio_folder_path, audio_file)
    segments, meta = model.transcribe(audio_file_path, language="es", beam_size = 5, best_of = 5, patience = 2)

    num_segments = len(list(segments))
    print(f"\nSegmentos generados para {audio_file}: {num_segments}")

    transcript_output_path = os.path.join(audio_folder_path, f"{os.path.splitext(audio_file)[0]}.txt")

    with open(transcript_output_path, "w") as file:
        for segment in segments:
            file.write(segment.text + "\n")
    print(meta)
    #print(f"Transcripción guardada exitosamente: {transcript_output_path}\n")

end_time = time.time()
execution_time = end_time - start_time
print("-"*30)
print(f"\nProceso de transcripción completado en {execution_time:.2f} segundos.")


Segmentos generados para Universidad Central - Deep Learning - Clase 2: Regularización de Redes Neuronales (2024-08-17 07:15 GMT-5).mp3: 713
TranscriptionInfo(language='es', language_probability=1, duration=4138.9106875, duration_after_vad=4138.9106875, all_language_probs=None, transcription_options=TranscriptionOptions(beam_size=5, best_of=5, patience=2, length_penalty=1, repetition_penalty=1, no_repeat_ngram_size=0, log_prob_threshold=-1.0, no_speech_threshold=0.6, compression_ratio_threshold=2.4, condition_on_previous_text=True, prompt_reset_on_temperature=0.5, temperatures=[0.0, 0.2, 0.4, 0.6, 0.8, 1.0], initial_prompt=None, prefix=None, suppress_blank=True, suppress_tokens=[-1], without_timestamps=False, max_initial_timestamp=1.0, word_timestamps=False, prepend_punctuations='"\'“¿([{-', append_punctuations='"\'.。,，!！?？:：”)]}、', max_new_tokens=None, clip_timestamps='0', hallucination_silence_threshold=None, hotwords=None), vad_options=None)

Segmentos generados para Universidad C

# Basic model with arbitrary hyperparameter adjustment + HW

In [18]:
audio_folder_path = "/content/drive/MyDrive/MAESTRIA/4 SEMESTRE/Deep Learning/data/audio/"

# 'tiny', 'base', 'small', 'medium', 'large'
model = WhisperModel("base", compute_type="float16")
hotwords = ' '.join(['inteligencia artificial', 'deep learning', 'aprendizaje profundo', 'gradiente', 'derivadas'])

audio_files = [f for f in os.listdir(audio_folder_path) if f.endswith('.mp3')]

start_time = time.time()

for audio_file in audio_files:
    audio_file_path = os.path.join(audio_folder_path, audio_file)
    segments, meta = model.transcribe(audio_file_path, language="es", beam_size = 5, best_of = 5, patience = 2, hotwords=hotwords)

    num_segments = len(list(segments))
    print(f"\nSegmentos generados para {audio_file}: {num_segments}")

    transcript_output_path = os.path.join(audio_folder_path, f"{os.path.splitext(audio_file)[0]}.txt")

    with open(transcript_output_path, "w") as file:
        for segment in segments:
            file.write(segment.text + "\n")
    print(meta)
    #print(f"Transcripción guardada exitosamente: {transcript_output_path}\n")

end_time = time.time()
execution_time = end_time - start_time
print("-"*30)
print(f"\nProceso de transcripción completado en {execution_time:.2f} segundos.")


Segmentos generados para Universidad Central - Deep Learning - Clase 2: Regularización de Redes Neuronales (2024-08-17 07:15 GMT-5).mp3: 959
TranscriptionInfo(language='es', language_probability=1, duration=4138.9106875, duration_after_vad=4138.9106875, all_language_probs=None, transcription_options=TranscriptionOptions(beam_size=5, best_of=5, patience=2, length_penalty=1, repetition_penalty=1, no_repeat_ngram_size=0, log_prob_threshold=-1.0, no_speech_threshold=0.6, compression_ratio_threshold=2.4, condition_on_previous_text=True, prompt_reset_on_temperature=0.5, temperatures=[0.0, 0.2, 0.4, 0.6, 0.8, 1.0], initial_prompt=None, prefix=None, suppress_blank=True, suppress_tokens=[-1], without_timestamps=False, max_initial_timestamp=1.0, word_timestamps=False, prepend_punctuations='"\'“¿([{-', append_punctuations='"\'.。,，!！?？:：”)]}、', max_new_tokens=None, clip_timestamps='0', hallucination_silence_threshold=None, hotwords='inteligencia artificial deep learning aprendizaje profundo grad

# Resultados

|| Basic Model | Basic Model (arbitrary HyperP) | Basic Model (arbitrary HyperP+HW) |
|---|---|---|---|
| Segments | 1440+1943 = 3383 | 713+1159 = 1872 | 959+838=1797 |
| Language P. | 98%~ |  1 | 1 |
| Time (seconds) | 344,81 | 263,28 | 220,4 |